config/manager/manager.yaml

```diff
@@ -43,6 +43,7 @@ spec:
         name: manager
         securityContext:
           allowPrivilegeEscalation: false
+          runAsUser: 1000
```

Comment:
we should be setting the user in the dockerfile itself like so:

```Dockerfile
USER 1000:1000
```

Reply:
sorry for the late reply,been travelling to attend Kubecon India 😅 ,i will make the changes

Reply:
Hi @aryan9600 ,i tried by placing `USER 1001:1001` in dockerfile,it shows the same error as `error: container has runasnonroot and image will run as root`

Reply:
you might be setting the user in the build step of the Dockerfile. you should set it in the final step of the Dockerfile: 

```Dockerfile
FROM --platform=$BUILDPLATFORM tonistiigi/xx AS xx-tools

FROM --platform=$BUILDPLATFORM rust:alpine AS builder

RUN apk add --no-cache clang lld

WORKDIR /workspace

COPY --from=xx-tools / /

ARG TARGETPLATFORM
ARG PROJECT_DIR=/workspace
ARG BUILD_DIR=$PROJECT_DIR/build

RUN --mount=type=bind,source=../controlplane/src/,target=src \
    --mount=type=bind,source=../controlplane/Cargo.toml,target=Cargo.toml \
    --mount=type=bind,source=../controlplane/Cargo.lock,target=Cargo.lock \
    xx-cargo build --release --target-dir $BUILD_DIR && \
    xx-verify ./build/$(xx-cargo --print-target-triple)/release/controller

RUN cp ./build/$(xx-cargo --print-target-triple)/release/controller /workspace/manager

FROM alpine:latest

WORKDIR /

USER 1000:1000
COPY --from=builder /workspace/manager /manager
```

i can get the application to run with this Dockerfile and without the need for `runAsUser` in the deployment configuration.

Reply:
hi @aryan9600 ,i did set in the final step but its not working 🥲 ,can we hop on a meet whenever you are free?https://calendly.com/mrrishi-kubefirst/30min